smithy-smoke-test-traits/src/main/resources/META-INF/smithy/smithy.test.smithy

```diff
@@ -0,0 +1,91 @@
+$version: "2.0"
+
+namespace smithy.test
+
+/// Defines a set of test cases to send to a live service to ensure that a
+/// client can successfully connect to a service and get the right kind of
+/// response.
+@trait(selector: "operation")
+list smokeTests {
+    /// A list of smoke tests to perform on the operation
+    member: SmokeTestCase
+}
+
+/// A single smoke test case definition.
+@private
+structure SmokeTestCase {
+    /// The identifier of the test case. This identifier may be used by
+    /// smoke test implementations to generate test case names. The provided
+    /// `id` MUST match Smithy's `IDENTIFIER` ABNF. No two test cases can
+    /// share the same ID, including test cases defined for other operations
+    /// bound to the same service.
+    @required
+    @pattern("^[A-Za-z_][A-Za-z0-9_]+$")
+    id: String
+
+    /// Defines the input parameters used to generate the request. These
+    /// parameters MUST be compatible with the input of the operation.
+    ///
+    /// Parameter values that contain binary data MUST be defined using values
```

Comment:
Is this consistent with protocol tests? Do we encode binary data that can't be represented this way with base64?

Reply:
This is consistent with protocol tests, yes: https://smithy.io/2.0/additional-specs/http-protocol-compliance-tests.html#smithy-test-httprequesttests-trait